pkg/i2gw/providers/apisix/resource_reader.go

```diff
@@ -39,7 +38,7 @@ func (r *resourceReader) readResourcesFromCluster(ctx context.Context) (*storage
 	// read apisix related resources from cluster.
 	storage := newResourcesStorage()
 
-	ingresses, err := common.ReadIngressesFromCluster(ctx, r.conf.Client, ApisixIngressClass)
+	ingresses, err := i2gw.ReadIngressesFromCluster(ctx, r.conf.Client, ApisixIngressClass)
```

Comment:
I needed to move the `resource-reader` files to the i2gw package to avoid the import cycle as with this change we read resources not only from the provider packages but from i2gw as well, as we want to read resources that do not have to be converted.